### ISSUE TITLE
Why do the distilled dataset contain continuous variables?

### ISSUE THREAD

**Issue by WanliYoung** [noveens/distill_cf#4]
Hello @noveens,
Thank you for your excellent work and patience response before! I'm confused about the distilled dataset saved in './results/distilled_data' seems to contain continuous variables not 0/1 variables. Could you please explain it to me? Thank you very much!

Here is the printed output：
[[0.08609356 0.05428863 0.04905649 ... 0.04361659 0.04393608 0.07505255]
 [0.06505322 0.05603834 0.07179898 ... 0.04712014 0.04364795 0.04961397]
 [0.07103189 0.0568559  0.04919334 ... 0.0483388  0.04975031 0.06032645]
 ...
 [0.04147852 0.03520147 0.03603516 ... 0.05840685 0.06582747 0.06773392]
 [0.05445674 0.05975632 0.08270465 ... 0.04501963 0.05601563 0.06489903]
 [0.05308461 0.0871274  0.06084298 ... 0.05301113 0.04530666 0.05996091]]

**Comment by noveens** [comment]
Hey @wending0417,

Thanks for your good question! 
if you think about the implicit feedback setup closely, you will see that even though the original data is discrete; it doesn't 
necessarily need to be discrete for training models. For example, in a recommendation setting of 8 items:
```
A user with viewing history: [0, 1, 0, 0, 1, 0, 0, 0] is used to train models like MVAE, EASE, or ∞-AE.
But a user with viewing history: [0.1, 0.8, 0.1, 0, 0.9, 0, 0.1, 0] can also be used to train the same models
```
To be flexible for this in our paper, the $\tau$ (temperature) hyper-parameter in Gumbel softmax controls the amount of `discrete-ness` in the synthesized data. The default value for $\tau=10$ will synthesize highly continuous values; but smaller values, e.g., $\tau \in [0.1, 1]$ will generate almost discrete data. 

Note that having a discrete data generation constraint (i.e., low $\tau$) will **limit** the flexibility in the data generation procedure typically leading to a lower data compression rate (i.e., need more data to train the same quality of models).

Does that make sense?

Cheers,
Noveen

**Comment by WanliYoung** [comment]
@noveens Thanks for your patient explanation! Actually, I have thought about that continuous values can be used to train models, because continuous values can also reflect the importance of items like [0.1, 0.8, 0.1, 0, 0.9, 0, 0.1, 0]. However, I checked the distilled dataset of ml-1m (500*3706), the maximum value is just about 0.10, is that correct? And I used the distilled dataset to train other models, its performance seems to be poor.  I'm not sure if it's due to the small values? Looking forward to your kind reply. Thank you very much!
